scrapyd_k8s/launcher/k8s.py

```diff
@@ -121,8 +135,65 @@ def cancel(self, project, job_id, signal):
             )
         return prevstate
 
-    def enable_joblogs(self, config):
-        joblogs_init(config)
+    def enable_joblogs(self, config, resource_watcher):
+        joblogs_init(config, resource_watcher)
+
+    def unsuspend_job(self, job_id: str):
+        job_name = self._get_job_name(job_id)
+        if not job_name:
+            logger.error(f"Cannot unsuspend job {job_id}: job name not found.")
+            return False
+        try:
+            self._k8s_batch.patch_namespaced_job(
+                name=job_name,
+                namespace=self._namespace,
+                body={'spec': {'suspend': False}}
+            )
+            logger.info(f"Job {job_id} unsuspended.")
+            return True
+        except Exception as e:
+            logger.exception(f"Error unsuspending job {job_id}: {e}")
+            return False
+
+    def get_running_jobs_count(self) -> int:
+        """
+        Returns the number of currently active (unsuspended, not completed, not failed) jobs.
+        """
+        label_selector = f"{self.LABEL_JOB_ID}"
+        jobs = self._k8s_batch.list_namespaced_job(
+            namespace=self._namespace,
+            label_selector=label_selector
+        )
+
+        active_jobs = []
+        for job in jobs.items:
+            job_name = job.metadata.name
+            is_suspended = job.spec.suspend
+            is_completed = job.status.completion_time is not None
+            has_failed = job.status.failed is not None and job.status.failed > 0
+            logger.debug(f"Job {job_name}: suspended={is_suspended}, completed={is_completed}, failed={has_failed}")
+
+            if not is_suspended and not is_completed and not has_failed:
+                active_jobs.append(job)
+
+        logger.debug(f"Active jobs: {[job.metadata.name for job in active_jobs]}")
+        logger.debug(f"Found {len(active_jobs)} active jobs.")
+
+        return len(active_jobs)
+
+    def list_suspended_jobs(self, label_selector: str = ""):
+        try:
+            jobs = self._k8s_batch.list_namespaced_job(
+                namespace=self._namespace,
+                label_selector=label_selector
+            )
+            # Filter jobs where spec.suspend == True
+            suspended_jobs = [job for job in jobs.items if job.spec.suspend]
+            logger.debug(f"Found {len(suspended_jobs)} suspended jobs.")
+            return suspended_jobs
+        except Exception as e:
+            logger.exception(f"Error listing suspended jobs: {e}")
+            return []
```

Comment:
Is there a situation where you would want to get both the running and suspended job count? Then it could be nice to do one call to `list_namespaced_job` to obtain both.

Reply:
Yes, good point, working on it

Reply:
So, I've been thinking about this, but those are two very different flows that require the latest cluster state when called, but I can make the code a bit more DRY.